src/components/layers/InteractivePointLayer/index.tsx

```diff
@@ -0,0 +1,286 @@
+
+      dragging.current = { pointID: id, offset };
+
+      onDragStart?.(id, offset, e);
+      e.preventDefault();
+    };
+
+    const handleMouseUp = (e: any) => {
+      if (dragging.current === null) return;
+      onDragEnd?.(dragging.current.pointID, e);
+      dragging.current = null;
+
+      if (lastHoverId.current !== null) {
+        onHoverLeave?.(lastHoverId.current, e);
+        lastHoverId.current = null;
+      }
+    };
+
+    // Events for click and hover handlers
+    map.on("click", pointLayerID, handleClick);
+    map.on("mousemove", pointLayerID, handleMouseMove);
+    map.on("mouseleave", pointLayerID, handleMouseMove);
+
+    // Events for drag handlers
+    map.on("mousedown", pointLayerID, handleMouseDown);
+    map.on("mousemove", handleDrag);
+    map.on("mouseup", handleMouseUp);
+
+    map.on("touchstart", pointLayerID, handleMouseDown);
+    map.on("touchmove", handleDrag);
+    map.on("touchend", handleMouseUp);
+    map.on("touchcancel", handleMouseUp);
+
+    // Make sure to capture pointerup events anywhere in the window
+    window.addEventListener("pointerup", handleMouseUp);
+
+    // Clean up events when we're done
+    return () => {
+      map.off("click", pointLayerID, handleClick);
+      map.off("mousemove", pointLayerID, handleMouseMove);
+      map.off("mouseleave", pointLayerID, handleMouseMove);
+
+      map.off("mousedown", pointLayerID, handleMouseDown);
+      map.off("mousemove", handleDrag);
+      map.off("mouseup", handleMouseUp);
+
+      map.off("touchstart", pointLayerID, handleMouseDown);
+      map.off("touchmove", handleDrag);
+      map.off("touchend", handleMouseUp);
+      map.off("touchcancel", handleMouseUp);
+
+      window.removeEventListener("pointerup", handleMouseUp);
+    };
+  }, [
+    map,
+    onClick,
+    onDrag,
+    onDragEnd,
+    onDragStart,
+    onHoverEnter,
+    onHoverLeave,
+    pointIndex,
+    pointLayerID,
+  ]);
+
+  return <PointLayer {...props} onAdd={setPointLayerID} />;
+};
+
+/** Given a Mapbox `MapMouseEvent`, sort any point features that the event
+ *  involves by their distance to the mouse in ascending order.
+ *
+ *  Mapbox doesn't have any native concept of event capturing, so the event it
+ *  fires will list out all the features that are hit. We most likely only want
+ *  to get the one closest to the mouse event.
+ */
+const getPointsSortedByDistance = (
+  e: (mapboxgl.MapMouseEvent | mapboxgl.MapTouchEvent) & {
+    features?: mapboxgl.MapboxGeoJSONFeature[] | undefined;
+  } & mapboxgl.EventData
+) => {
+  const features = e.features;
+  if (!features) return null;
+  const eventPoint = e.lngLat;
+
+  return features
+    .map((f) => ({
+      ...f,
+      id: (f.properties?.["id"] as number | string) ?? null,
+      geometry: f.geometry, // This is a getter and needs to be manually copied
+      distance:
+        f.geometry.type === "Point"
+          ? eventPoint.distanceTo(
```

Comment:
Since there is no need for the actual distance, only relative distance, you could get away with a less computationally intensive distance proxy than actual euclidian distance, which requires multiplication and square roots.

Reply:
(Since Manhattan and euclidean distance are increasing and decreasing functions within the same intervals, I think Manhattan is just as useful for sorting https://www.geogebra.org/3d/bze238fz)

Reply:
I am realizing, I think that these are all features that overlap and would thus be returned by the map after a map event. If that is the case, then the optimizations I suggested are very unlikely to be needed.

Reply:
That's a great point— I hadn't considered cheaper distance functions for sorting. But as you say in your last comment, this is only sorting the overlapping features (which will be at most a half dozen or so in normal use). I think I'll leave it as is if only because this will gracefully handle the edge case at the antimeridian (where e.g. euclidean distance would fail) and performance is pretty good.